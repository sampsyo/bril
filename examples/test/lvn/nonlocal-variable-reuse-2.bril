# CMD: bril2json < {filename} | python3 ../../lvn.py -p | bril2txt
@main {
  a: int = const 42;
.lbl:
  # While `a` is outside the local scope, we essentially
  # have a no-op here, so nonlocal re-naming isn't necessary.
  a: int = id a;
  a: int = const 5;
  print a;
}
