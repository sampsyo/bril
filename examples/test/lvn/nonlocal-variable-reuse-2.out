@main {
  a: int = const 42;
.lbl:
  a: int = id a;
  a: int = const 5;
  print a;
}
